File: BAEKJOON/java/3613_Java_vs_C++.java
import java.io.*;

public class Main {
    static String s;

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        s = br.readLine();
    }

    private static boolean checkJava(String s) {
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) >= 'A' && s.charAt(i) <= 'Z') {
                return true;
            }
        }

        return false;
    }

    private static boolean checkCpp(String s) {
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '_') {
                return true;
            }
        }

        return false;
    }

    private static boolean errorCheck(String s) {
        boolean isJava = checkJava(s);
        boolean isCpp = checkCpp(s);

        if (isJava && isCpp) {
            return false;
        }

        if ((s.charAt(0) == '_') || (s.charAt(0) >= 'A' && s.charAt(0) <= 'Z')) {
            return false;
        }

        if (s.charAt(s.length() - 1) == '_') {
            return false;
        }

        for (int i = 0; i < s.length() - 1; i++) {
            if (s.charAt(i) == s.charAt(i + 1) && s.charAt(i) == '_') {
                return false;
            }
        }

        return true;
    }

    private static String makeCpp(String s) {
        StringBuilder ret = new StringBuilder();

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) >= 'A' && s.charAt(i) <= 'Z') {
                ret.append("_");
                ret.append(Character.toLowerCase(s.charAt(i)));
            } else {
                ret.append(s.charAt(i));
            }
        }

        return ret.toString();
    }

    private static String makeJava(String s) {
        StringBuilder ret = new StringBuilder();

        boolean up = false;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '_') {
                up = true;
            } else {
                if (up) {
                    ret.append(Character.toUpperCase(s.charAt(i)));
                    up = false;
                } else {
                    ret.append(s.charAt(i));
                }
            }
        }

        return ret.toString();
    }

    private static void solution() throws IOException {
        if (!errorCheck(s)) {
            bw.write("Error!");
            return;
        }

        if (checkJava(s)) {
            bw.write(makeCpp(s));
            return;
        }

        bw.write(makeJava(s));
    }
}

File: BAEKJOON/cpp/3613_Java_vs_C++.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

string s;

void input() {
	cin >> s;
}

bool checkJava(string s) {
	for (int i = 0; i < s.length(); i++) {
		if (isupper(s[i])) {
			return true;
		}
	}

	return false;
}

bool checkCpp(string s) {
	for (int i = 0; i < s.length(); i++) {
		if (s[i] == '_') {
			return true;
		}
	}

	return false;
}

bool errorCheck(string s) {
	bool isJava = checkJava(s);
	bool isCpp = checkCpp(s);

	if (isJava && isCpp) {
		return false;
	}

	if ((s[0] == '_') || isupper(s[0])) {
		return false;
	}

	if (s[s.length() - 1] == '_') {
		return false;
	}

	for (int i = 0; i < s.length() - 1; i++) {
		if (s[i] == s[i + 1] && s[i] == '_') {
			return false;
		}
	}

	return true;
}

string makeCpp(string s) {
	string ret;

	for (int i = 0; i < s.length(); i++) {
		if (isupper(s[i])) {
			ret += "_";
			ret += tolower(s[i]);
		}
		else {
			ret += s[i];
		}
	}

	return ret;
}

string makeJava(string s) {
	string ret;

	bool up = false;
	for (int i = 0; i < s.length(); i++) {
		if (s[i] == '_') {
			up = true;
		}
		else {
			if (up) {
				ret += toupper(s[i]);
				up = false;
			}
			else {
				ret += s[i];
			}
		}
	}

	return ret;
}

void solution() {
	if (!errorCheck(s)) {
		cout << "Error!";
		return;
	}

	if (checkJava(s)) {
		cout << makeCpp(s);
		return;
	}

	cout << makeJava(s);
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}